strutil/cmdline_test.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+type cmdlineTestSuite struct{}
+
+var _ = Suite(&cmdlineTestSuite{})
+
+func (s *cmdlineTestSuite) TestSplitKernelCommandLine(c *C) {
+	for idx, tc := range []struct {
+		cmd    string
+		exp    []string
+		errStr string
+	}{
+		{cmd: ``, exp: nil},
+		{cmd: `foo bar baz`, exp: []string{"foo", "bar", "baz"}},
+		{cmd: `foo=" many   spaces  " bar`, exp: []string{`foo=" many   spaces  "`, "bar"}},
+		{cmd: `foo="1$2"`, exp: []string{`foo="1$2"`}},
+		{cmd: `foo=1$2`, exp: []string{`foo=1$2`}},
+		{cmd: `foo= bar`, exp: []string{"foo=", "bar"}},
+		{cmd: `foo= bar`, exp: []string{"foo=", "bar"}},
```

Comment:
43 and 44 look identical to me, if they are not, maybe a comment why they are not?

Reply:
Heh, too much rebasing